File: bot-api/python/src/robocode_tank_royale/bot_api/team_message_batch.py
from dataclasses import dataclass
from typing import Any, Iterable


@dataclass(frozen=True)
class TeamMessageBatch:
    """Immutable ordered payloads delivered as one team-message event on the next turn.

    A batch must contain at least one non-null value. Each entry counts toward the turn's logical-payload limit.
    """
    messages: tuple[Any, ...]

    def __init__(self, messages: Iterable[Any]):
        # Materialize first so a one-shot iterable (e.g. a generator) is validated and stored intact
        items = tuple(messages)
        if not items:
            raise ValueError("A team message batch must contain at least one message")
        if any(message is None for message in items):
            raise ValueError("A team message batch cannot contain null messages")
        object.__setattr__(self, "messages", items)

File: bot-api/java/src/main/java/dev/robocode/tankroyale/botapi/TeamMessageBatch.java
package dev.robocode.tankroyale.botapi;

import java.util.Collection;
import java.util.List;

/**
 * An immutable, ordered group of non-null payloads delivered through one team-message event on the next turn.
 * Use {@link IBaseBot#broadcastTeamMessageBatch(java.util.Collection)} or
 * {@link IBaseBot#sendTeamMessageBatch(int, java.util.Collection)} to enqueue a batch.
 */
public final class TeamMessageBatch {
    private final List<Object> messages;

    public TeamMessageBatch(Collection<?> messages) {
        if (messages == null || messages.isEmpty()) {
            throw new IllegalArgumentException("A team message batch must contain at least one message");
        }
        if (messages.stream().anyMatch(message -> message == null)) {
            throw new IllegalArgumentException("A team message batch cannot contain null messages");
        }
        this.messages = List.copyOf(messages);
    }

    public List<Object> getMessages() {
        return messages;
    }
}
